api-inference-community/api_inference_community/validation.py

```diff
@@ -161,9 +160,18 @@ def check_params(params, tag):
 def check_inputs(inputs, tag):
     if tag in INPUTS_MAPPING:
         INPUTS_MAPPING[tag].parse_obj(inputs)
+    elif tag in BATCH_ENABLED__PIPELINES:
+        if isinstance(inputs, list):
+            if len(inputs) == 0:
+                raise ValueError(
+                    "The inputs is invalid, at least one input is required"
+                )
+            if not all(isinstance(input, str) for input in inputs):
```

Comment:
I would actually vote to add the real validation in a `INPUTS_MAPPING` validator and remove `BATCH_ENABLED_PIPELINE` and the rest.

Wdyt ?

Reply:
I think this case is a bit different.

Pydantic base model uses `parse_obj` to parse a dictionary. But for pipelines with batching, we really get a list or a string, so `parse_obj` would simply fail.

Reply:
```python

from pydantic import BaseModel                                                    
from typing import List, Union                                                    
                                                                                  
                                                                                  
class Input(BaseModel):                                                           
    inputs: Union[List[str], str]                                                 
                                                                                  
    def __iter__(self):                                                           
        return iter(self.inputs)                                                  
                                                                                  
    def __getitem__(self, item):                                                  
        return self.inputs[item]                                                  
                                                                                  
                                                                                  
print(Input(inputs=["A", "B"]).inputs)                                            
print(Input(inputs="A").inputs)                                                   
print(Input(inputs=["A", 5]).inputs)                                              
print(Input.parse_obj({"inputs": ["A", "B"]}).inputs)    

```
What do you think about this approach ?

Maybe we could also remove the else branch for raw strings at the same time.

It could be left for another PR, but I think making validation consistent and in a single place (`INPUTS_MAPPING`) is probably going to help rather than hurt.

Reply:
Other option is `__root__` (seems very similar to naming things `inputs` to me) https://pydantic-docs.helpmanual.io/usage/models/#custom-root-types

Reply:
I don't think the first approach is viable here since we still use `parse_obj` which would not work - doing separate parsing would require two lists which is what we're trying to avoid.

Anyways, the `__root__` approach seemed to work very well so I took the opportunity to move all the pipelines to this validation form.